crates/macro/src/html_tree/html_component.rs

```diff
@@ -21,11 +21,9 @@ pub struct HtmlComponent {
 
 impl PeekValue<()> for HtmlComponent {
     fn peek(cursor: Cursor) -> Option<()> {
-        let (punct, cursor) = cursor.punct()?;
-        (punct.as_char() == '<').as_option()?;
-
-        HtmlComponent::peek_type(cursor)?;
-        Some(())
+        HtmlComponentOpen::peek(cursor)
```

Comment:
I would like all of the changes in this file to be moved to a new PR. Looks like the main purpose is to have better error messages for this:

```

error: expected type, found `/`
  --> $DIR/html-component-fail.rs:74:14
   |
74 |     html! { </Child> };
   |              ^ expected type
```
```
	error: this close tag has no corresponding open tag
	  --> $DIR/html-component-fail.rs:74:13
	   |
	74 |     html! { </Child> };
	   |             ^^^^^^^^
```

Reply:
Correct. I'll separate this stuff.

Reply:
Wait, so I had to fix this otherwise the unittests were failing. You want to merge the fix first before this PR?